### PR TITLE
Spring boot package [GC - data stream]

### PR DESCRIPTION
**Pull request by sunny-elastic**
## What does this PR do?

- Generated the skeleton of Spring Boot integration package.
- Added **GC** data stream
- Added data collection logic.
- Added the ingest pipelines.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added system test cases.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that GC data stream collect metrics.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^8.0.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/spring_boot directory.
- Run the following command to run tests. 
> elastic-package test 

Note: We have covered dashboards and the visualisations for all data streams of spring boot into separate PR. Also Kibana version will be updated to 8.1.0 in manifest.yml after testing this integration on 8.1.0.

### CODE REVIEW

File: packages/spring_boot/data_stream/gc/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+title: GC Metrics
```
Comment (mtojek) [inline packages/spring_boot/data_stream/gc/manifest.yml:1]: Let's reveal GC abbr. at least for titles and headlines.

File: packages/spring_boot/data_stream/gc/sample_event.json [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        "name": "jmx",
+        "period": 60000
+    },
+    "service": {
+        "address": "http://springboot:8090/actuator/jolokia",
+        "type": "jolokia"
+    },
+    "spring_boot": {
+        "gc": {
+            "garbage_collector": {
```
Comment (mtojek) [inline packages/spring_boot/data_stream/gc/sample_event.json:65]: Please merge these levels into `spring_boot.gc.*`.

File: packages/spring_boot/data_stream/gc/sample_event.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                "collection": {
```
Comment (mtojek) [inline packages/spring_boot/data_stream/gc/sample_event.json:66]: What does it mean? Should we skip it?

File: packages/spring_boot/data_stream/gc/sample_event.json [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                    "count": 2,
+                    "time": 107
+                },
+                "last_gc_info": {
```
Comment (mtojek) [inline packages/spring_boot/data_stream/gc/sample_event.json:70]: I reviewed a similar GC structure for Spark. Could you please make it consistent with other integrations? PR: https://github.com/elastic/integrations/pull/2811/files#r837067908

File: packages/spring_boot/data_stream/gc/manifest.yml (hunk 1 above)
Comment (sunny-elastic) [reply packages/spring_boot/data_stream/gc/manifest.yml]: Yes added.

File: packages/spring_boot/data_stream/gc/sample_event.json (hunk 2 above)
Comment (sunny-elastic) [reply packages/spring_boot/data_stream/gc/sample_event.json]: Yes redefined

File: packages/spring_boot/data_stream/gc/sample_event.json (hunk 3 above)
Comment (sunny-elastic) [reply packages/spring_boot/data_stream/gc/sample_event.json]: Yeah we updated this structure to ```spring_boot.gc.collection```

File: packages/spring_boot/data_stream/gc/sample_event.json (hunk 4 above)
Comment (sunny-elastic) [reply packages/spring_boot/data_stream/gc/sample_event.json]: ok so this structure is different here. here ```last_gc_info``` is an object.

File: packages/spring_boot/data_stream/gc/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      field: jolokia.metrics.name
+      target_field: spring_boot.gc.name
+      ignore_missing: true
+      ignore_failure: true            
+  - remove:
+      field: jolokia
+      ignore_failure: true
+      ignore_missing: true
+on_failure:
+  - set:
```
Comment (yug-elastic) [inline packages/spring_boot/data_stream/gc/elasticsearch/ingest_pipeline/default.yml:117]: Is this indentation correct?

**Review by yug-elastic — COMMENTED** [review]
Just a couple of nitpicks and some things to clarify, LGTM otherwise.
Please wait for a review from others as well.

File: packages/spring_boot/_dev/build/docs/README.md [hunk 6]
```diff
@@ -69,3 +69,13 @@ This is the `threading` data stream.
 {{event "threading"}}
 
 {{fields "threading"}}
+
+### GC Metrics
+
+This is the `gc` data stream.
+
+- This data stream gives metrics related to Garbage Collector(GC) Memory.
```
[the remark this answers is not in the document]
Comment (yug-elastic) [reply packages/spring_boot/_dev/build/docs/README.md:77]: Please make the same change everywhere applicable.

File: packages/spring_boot/data_stream/gc/elasticsearch/ingest_pipeline/default.yml (hunk 5 above)
Comment (yug-elastic) [reply packages/spring_boot/data_stream/gc/elasticsearch/ingest_pipeline/default.yml:117]: I tried it out, it's working fine :)

Comment (sunny-elastic) [reply packages/spring_boot/data_stream/gc/elasticsearch/ingest_pipeline/default.yml:117]: Yes thanks

File: packages/spring_boot/_dev/build/docs/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (sunny-elastic) [reply packages/spring_boot/_dev/build/docs/README.md:77]: Its done

**Review by yug-elastic — APPROVED** [review]
LGTM, let's wait for @mtojek's re-review over the same.

File: packages/spring_boot/_dev/build/docs/README.md (hunk 6 above)
Comment (mtojek) [inline packages/spring_boot/_dev/build/docs/README.md:77]: nit: to Garbage Collector (GC).

**Review by mtojek — APPROVED** [review]
LGTM!

File: packages/spring_boot/_dev/build/docs/README.md (hunk 6 above)
Comment (sunny-elastic) [reply packages/spring_boot/_dev/build/docs/README.md]: Fixed

**Review by mtojek — APPROVED** [review]

**Pull request merged**